src/mlpack/core/optimizers/ada_delta/ada_delta_impl.hpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file ada_delta_impl.hpp
+ * @author Ryan Curtin
+ * @author Vasanth Kalingeri
+ * @author Abhinav Moudgil
+ *
+ * Implementation of the AdaDelta optimizer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADADELTA_ADA_DELTA_IMPL_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADADELTA_ADA_DELTA_IMPL_HPP
+
+#include "ada_delta.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+template<typename DecomposableFunctionType>
+AdaDelta<DecomposableFunctionType>::AdaDelta(DecomposableFunctionType& function,
+                                           const double stepSize,
```

Comment:
Pedantic style comment, please align the DecomposableFunctionType with the other function parameters.